frontend/test/pytest/test_measurement_primitives.py

```diff
@@ -48,10 +213,158 @@ def f():
+    %0 = "quantum.compbasis"() : () -> !quantum.obs
+    %1:2 = "quantum.counts"(%0) : (!quantum.obs) -> (tensor<1xf64>, tensor<1xi64>)
+    return %1#0, %1#1 : tensor<1xf64>, tensor<1xi64>
+  }
+}
+"""
+    )
+
+    # assert jaxpr.eqns[1].primitive == counts_p
+    # assert jaxpr.eqns[1].params == {"shape": (1,), "shots": 5}
+    # assert jaxpr.eqns[1].outvars[0].aval.shape == (1,)
+    # assert jaxpr.eqns[1].outvars[1].aval.shape == (1,)
+
+
+@pytest.mark.xfail(reason="[WIP] Convert to lit test")
+def test_counts_dynamic_shape():
+    """Test that the counts primitive can be captured by jaxpr."""
+
+    def f(shots):
+        obs = compbasis_p.bind()
+        return counts_p.bind(obs, shots, shape=(1,))
+
+    jaxpr = jax.make_jaxpr(f)(5)
+    mlir = lower_jaxpr_to_mlir(jax.make_jaxpr(f)(5), "foo")[0]
+    print(jaxpr)
+    print(mlir)
+    assert (
+        jaxpr
+        == """
+{ lambda ; a:i64[]. let
+    b:AbstractObs(num_qubits=0,primitive=compbasis) = compbasis
+    c:f64[1] d:i64[1] = counts[shape=(1,)] b a
+  in (c, d) }
+"""
+    )
+
+    assert (
+        mlir
+        == """
+module @foo {
+  func.func public @jit_foo(%arg0: tensor<i64>) -> (tensor<1xf64>, tensor<1xi64>) {
+    %0 = "quantum.compbasis"() : () -> !quantum.obs
+    %1:2 = "quantum.counts"(%0) : (!quantum.obs) -> (tensor<1xf64>, tensor<1xi64>)
+    return %1#0, %1#1 : tensor<1xf64>, tensor<1xi64>
+  }
+}
+"""
+    )
+
+    # assert jaxpr.eqns[1].primitive == counts_p
+    # assert jaxpr.eqns[1].params == {"shape": (1,), "shots": 5}
+    # assert jaxpr.eqns[1].outvars[0].aval.shape == (1,)
+    # assert jaxpr.eqns[1].outvars[1].aval.shape == (1,)
+
+
+def test_new_countsop_still_good_with_backend():
+    """Test that a `counts` program with dynamic shots can be executed correctly."""
+
+    @catalyst.qjit(keep_intermediate=True)
+    def workflow_dyn_counts(shots):
+        # qml.device still needs concrete shots
+        device = qml.device("lightning.qubit", wires=1, shots=10)
+
+        @qml.qnode(device)
+        def circuit():
+            qml.RX(1.5, 0)
+            return qml.counts()
+
+        return circuit()
+
+    workflow_dyn_counts(10)
+    old_ir = get_compilation_stage(workflow_dyn_counts, "mlir")
+    new_ir = old_ir.replace(
+        "catalyst.launch_kernel @module_circuit::@circuit() : () ->",
+        "catalyst.launch_kernel @module_circuit::@circuit(%arg0) : (tensor<i64>) ->",
+    )
+    new_ir = new_ir.replace(
+        "func.func public @circuit() ->", "func.func public @circuit(%arg0: tensor<i64>) ->"
+    )
+    new_ir = new_ir.replace(
+        "quantum.device shots(%extracted) [",
+        """%shots = tensor.extract %arg0[] : tensor<i64>
+      quantum.device shots(%shots) [""",
+    )
+
+    replace_ir(workflow_dyn_counts, "mlir", new_ir)
+    res = workflow_dyn_counts(4000)
+    print("after: ", res)
+    assert res[1][0] + res[1][1] == 4000
+
+    shutil.rmtree("workflow_dyn_counts")
+    shutil.rmtree("workflow_dyn_counts_1")
```

Comment:
[We have this method.](https://github.com/PennyLaneAI/catalyst/blob/845851662af490a45bff92ec32b5d1aaef7e7b10/frontend/catalyst/utils/filesystem.py#L49) If it is insufficient consider changing it.

Reply:
Ahh, [like here?](https://github.com/PennyLaneAI/catalyst/blob/1472b4d958219987e5d4ae55d14c10bb70a31c2d/frontend/test/pytest/test_debug.py#L245)

Reply:
@paul0403 yes

Reply:
done